Add krb tests

vakwetu opened the pull request.

vakwetu committed:
Add test to see if keytab is usable

vakwetu committed:
Added check for novajoin kerberos keytab

Re-ordered the executions so that we get some global variables
we need for the novajoin tests.

vakwetu committed:
Added check if novajoin running on non-containerized undercloud

Also check to see if the movajoin-server is reachable

vakwetu committed:
Fix location of keytab for containerized undercloud

JAORMX commented on tasks/main.yml:
```diff
@@ -1,11 +1,11 @@
 ---
+- name: Common tasks
+  import_tasks: common.yml
+
```
any reason why this is now ran in the beginning?

vakwetu replied:
Yes -- there is stuff that is defined in here -- things like the realm and other stuff in the ipa.conf that is used later on.
Easiest to run this first rather than repeat ourselves.

JAORMX commented on tasks/undercloud.yml:
```diff
@@ -1,4 +1,22 @@
+
+- name: Is the undercloud containerized?
+  set_fact:
+    containerized_undercloud: "{{ ansible_facts.packages['puppet-keystone'][0].version is version('13', '>=') }}"
```
Does this work upstream too?

vakwetu replied:
Not sure actually ... Do you have another test instead?  Look for the keystone container?

JAORMX commented on tasks/undercloud.yml:
```diff
@@ -146,3 +164,224 @@
+
+# We need to lookup stuff from the novajoin join.conf file but its
+# only readable by root.  make a temporary copy here, and read everything
+# we need.
```
can't we address this by adding become: true?

vakwetu replied:
No, it doesn't work - I tried.  Lookups always run in the context of the user running the playbook.

JAORMX commented on tasks/undercloud.yml:
```diff
@@ -146,3 +164,224 @@
+        principal_usable_status: '{{ helper_status_ok }}'
+        principal_usable_reason: "The principal {{ novajoin_principal }} is able to obtain a kerberos ticket"
+        principal_usable_recommendations: null
+        temp_krb_caches: "{{ temp_krb_caches + [ '/tmp/my_novajoin_krb5_ccache' ] }}"
```
oho! I didn't know you could do this!

vakwetu replied:
yup  -- all variables are global, and handlers always run at the end.

JAORMX commented:
By the way, the PR is failing CI because of ansible lint issues: https://travis-ci.org/JAORMX/tripleo-tls-everywhere-helper/jobs/512650447 you can see the job failure by clicking the "Details" link.

vakwetu commented:
Ack -cool -- will fix.

vakwetu committed:
New fixes to do things correctly with containers

vakwetu committed:
Restructure the tree

add_host is used to add containers to the inentory dynamically so
that we can treat the container and non-container novajoins in the
same way, rather than writing yucky container specific code.

add_host always runs on the controller host, but it selects a host
seemingly at random to be the host it "runs" on, and then bypasses
the rest of the host loop.  If you're lucky, the host it chooses is
the undercloud, and all goes well.  If you're unlucky (more likely
than not), the host it chooses is not the undercloud, and because of
when: conditions, the task is skipped.

The best way to deal with this to break things up into separate roles.
This is what this patch currently does.  We now have three roles -
tls-everywhere-helper-common, tls-everywhere-helper-overcloud and
tls-everywhere-helper-undercloud.

A checkin to remove the old code will follow this one.

vakwetu committed:
Clean up old structure

vakwetu committed:
Fix references to reportentry.yaml

vakwetu committed:
Add reportentry.yaml and split pre and post steps for undercloud

vakwetu committed:
Make the linter happy

vakwetu committed:
Update the README to describe new invocation

vakwetu committed:
Add check for novajoin_notifier and novajoin_server up for containers

vakwetu committed:
Update the meta files for Galaxy packaging
